internal/services/pam/pam.go

```diff
@@ -248,6 +248,10 @@ func (s Service) IsAuthenticated(ctx context.Context, req *authd.IARequest) (res
 			return nil, fmt.Errorf("user data from broker invalid: %v", err)
 		}
 
+		if err := s.checkIDConflicts(user); err != nil {
```

Comment:
I would do that directly on user and group creation rather, to not do it on every single login.

Reply:
Also if done at creation time, maybe we can just retry generating another ID?

Or not sure if that's changed now where the username plays a more important role

Reply:
We discussed about generating another ID but decided against it due to the possibility of the same user having different UIDs on different machines, which could create confusion.